src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -900,3 +915,11 @@ def _create_identity_instance(cli_ctx, *args, **kwargs):
     return Identity(*args, encrypt=encrypt, use_msal_http_cache=use_msal_http_cache,
                     enable_broker_on_windows=enable_broker_on_windows,
                     instance_discovery=instance_discovery, **kwargs)
+
+
+def _msal_managed_identity():
+    # Azure Arc
+    if "IDENTITY_ENDPOINT" in os.environ and "IMDS_ENDPOINT" in os.environ:
+        logger.debug("Azure Arc detected")
+        return True
```

Comment:
These env vars may not be present on Linux servers, so MSAL introduced a new detection method: https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/731.

After a new MSAL is released, we should switch to `_get_arc_endpoint`.

Reply:
Can we update to use `_get_arc_endpoint` now since new MSAL has been released for some time